woocommerce/woocommerce-android
# Order Creation: adding customer note

Closes: #5227 
Closes: #5228 

### Description
This PR handles the customer note section, it's based on the work done on #5405, but if the team decides against this component, the migration won't be hard.

I took liberty to add classes for managing the navigation: `OrderCreationNavigationTarget` and `OrderCreationNavigator`

### Testing instructions
1. Make sure the Order Creation experimental flag is activated
2. Go to the Order List and hit the FAB, select the Order Creation option if asked
3. Click on Add note button.
4. Make changes and make sure the "done" button appears as expected.
5. Click on Done and confirm the changes are saved.
6. Click on Edit and confirm it works. 

### Images/gif
| | | |
|- | - | -|
| <img width="400" alt="Screen Shot 2021-12-03 at 18 00 14" src="https://user-images.githubusercontent.com/1657201/144642602-d17cd603-d87f-41ee-bb97-4ebe01b126e5.png"> | <img width="400" alt="Screen Shot 2021-12-03 at 18 03 24" src="https://user-images.githubusercontent.com/1657201/144642945-8f4e9b9a-1ce8-4474-87f8-ef0176d001ad.png"> | <img width="400" alt="Screen Shot 2021-12-03 at 18 00 30" src="https://user-images.githubusercontent.com/1657201/144642596-50c924a0-037c-4526-b920-81f6f42dc315.png"> |
